tests/ui/structs/method-chain-expression-failure.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0599]: no method named `foo` found for struct `D` in the current scope
+  --> $DIR/method-chain-expression-failure.rs:24:19
+   |
+LL | struct D;
+   | -------- method `foo` not found for this struct
+...
+LL |     A.b().c().d().foo();
+   |                   ^^^ method not found in `D`
```

Comment:
Assuming that both B and C have a foo method, I think the ideal output to strive for would be something like

```
error[E0599]: no method named `foo` found for struct `D` in the current scope
  --> $DIR/method-chain-expression-failure.rs:24:19
   |
LL | struct D;
   | -------- method `foo` not found for this struct
...
LL |     A.b().c().d().foo();
   |           --- --- ^^^ method not found in `D`
   |           |   |
   |           |   method `foo` is available on `C`
   |           method `foo` is available on `B`
note: method `foo` is available on `B` and `C`
   |
   |     fn foo(&self) {}
   |        ^^^ method available on `B` here
...
   |     fn foo(&self) {}
   |        ^^^ method available on `C` here
help: you might have meant to call `foo` earlier in the method chain
   |
LL -     A.b().c().d().foo();
LL +     A.b().c().foo();
   |
LL -     A.b().c().d().foo();
LL +     A.b().foo();
   |
```

Going out of our way we would also check if `.d()` could be called after `.foo()` and be more accurate with that, but again, that's not something we need to accomplish all at once. I'm more than happy if all we have is the new span_labels on the main message.

Reply:
@estebank How do I add the `LL-`,  `LL+` comments? I think I can collect all the possible candidates and create the `note` and `help`